ast/src/parsed/asm.rs

```diff
@@ -91,6 +91,7 @@ pub enum DebugDirective {
 pub enum RegisterFlag {
     IsPC,
     IsAssignment,
+    IsReadOnly,
```

Comment:
What's a readonly register?

Reply:
https://github.com/powdr-labs/powdr/blob/call-to-vm/book/src/asm/registers.md#read-only-registers

Reply:
I don't quite get why they're called `readonly`. If they're used for inputs, they're also sort of "written" from the caller machine?

Reply:
Right, they are read-only within a block: they get set non-deterministically at the beginning of each block, and then they cannot change within this block. Any better name? `FunctionInput` ?